protocol/x/clob/keeper/orders.go

```diff
@@ -71,22 +71,24 @@ func (k Keeper) CancelShortTermOrder(
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), metrics.CancelShortTermOrder, metrics.Latency)
 	telemetry.IncrCounter(1, types.ModuleName, metrics.CancelShortTermOrder, metrics.Count)
 
-	// Perform all stateful validation on the order. Order may be stateful or short term.
+	// Perform all stateful validation on the short term order.
 	if err := k.PerformOrderCancellationStatefulValidation(ctx, msgCancelOrder, nextBlockHeight); err != nil {
 		return err
 	}
 
-	// Update in-memory orderbook to remove order.
+	// Update in-memory orderbook to remove short term order.
 	offchainUpdates, err := k.MemClob.CancelOrder(ctx, msgCancelOrder)
-
-	if err == nil {
-		k.sendOffchainMessagesWithTxHash(
-			offchainUpdates,
-			tmhash.Sum(ctx.TxBytes()),
-			metrics.SendCancelOrderOffchainUpdates,
-		)
+	if err != nil {
+		return err
 	}
-	return err
+
+	k.sendOffchainMessagesWithTxHash(
+		offchainUpdates,
+		tmhash.Sum(ctx.TxBytes()),
+		metrics.SendCancelOrderOffchainUpdates,
+	)
+
+	return nil
```

Comment:
Is `nil` return necessary in go? Been a while but I thought it wasn't needed

Reply:
`missing returncompiler[MissingReturn](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#MissingReturn)`